pkg/config/resource.go

```diff
@@ -690,12 +692,6 @@ func (m SchemaElementOptions) SetAddToObservation(el string) {
 	m[el].AddToObservation = true
 }
 
-// AddToObservation returns true if the schema element at the specified path
-// should be added to the CRD type's Observation type.
-func (m SchemaElementOptions) AddToObservation(el string) bool {
```

Comment:
I agree with this statement in the PR description about this code change:

> config.SchemaElementOptions.AddToObservation() has been removed due to its confusing naming. Again, not expected to cause migration issues for provider authors.

However, since this is part of our public configuration API (although I didn't observe the usage of the big three providers, and it will not have a significant effect), we may consider deprecating it first and removing it in the next releases.

Reply:
Done. Added the function back with a deprecation notice. This is an internal utility probably not used elsewhere but it makes sense to obey the API lifecycle management hygiene :) Maybe we should also consider a larger refactoring for future to move similar things to internal packages.